src/lib/systemd_journal/query.rb

```diff
@@ -40,70 +35,74 @@ class Query
     end
 
-    # String with a list of arguments for journalctl
-    def journalctl_args
-      return @journalctl_args if @journalctl_args
+    # Hash of options in the format expected by SystemdJournal::Journalctl
+    def journalctl_options
+      return @journalctl_options if @journalctl_options
 
-      args = []
+      @journalctl_options = {}
 
       # If a interval was specified, translate it to journalctl arguments
       if @interval
         case @interval
         when Array
-          args << time_argument(:since, interval[0])
-          args << time_argument(:until, interval[1])
+          @journalctl_options["since"] = @interval[0]
+          @journalctl_options["until"] = @interval[1]
         when Hash
-          args << time_argument(:since, interval[:since])
-          args << time_argument(:until, interval[:until])
+          @journalctl_options["since"] = @interval[:since]
+          @journalctl_options["until"] = @interval[:until]
         else
-          args << "--boot=\"#{@interval}\""
+          @journalctl_options["boot"] = @interval
         end
       end
       # Remove empty time arguments
-      args.compact!
-
-      # Add filters
-      FILTERS.each do |filter|
-        # Make sure it's an array and remove nils
-        values = [@filters[filter[:name]]].flatten.compact
-        values.each do |value|
-          args << "#{filter[:arg]}\"#{value}\""
-        end
-      end
-      
-      @journalctl_args = args.join(" ")
+      @journalctl_options.reject! {|k,v| v.nil? }
+
+      # Add filters...
+      @journalctl_options.merge!(@filters)
+      # ...expect 'match'
+      @journalctl_options.delete("match")
+
+      @journalctl_options
+    end
+
+    # Array of matches in the format expected by SystemdJournal::Journalctl
+    def journalctl_matches
+      @filters["match"]
     end
 
     # Calls journalctl and returns an Array of Entry objects
     def entries
-      Entry.all(journalctl_args)
+      Entry.all(options: journalctl_options, matches: journalctl_matches)
     end
 
     def to_s
       "<interval: #{@interval}, filters: #{@filters}>"
     end
 
-  private
-
-    # String representation of a time-based journalctl argument
-    # 
-    # @param arg [#to_s] name of the argument
-    # @param value [#strftime,#to_s]
-    def time_argument(arg, value)
-      return nil if value.nil?
-      if value.respond_to?(:strftime)
-        value = "#{value.strftime(JOURNALCTL_TIME_FORMAT)}"
+    # Array of system's boots registered in the journal.
+    #
+    # Each boot is represented by a hash with three elements, with all the keys
+    # being symbols and all the values being strings.
+    #  * id: 32-character identifier
+    #  * offset: offset relative to the current boot
+    #  * timestamps: timestamps of the first and last message for the boot
+    def self.boots
+      Journalctl.new({"list-boots" => nil}, nil).output.lines.map do |line|
+        next unless line.strip =~ /^ *(-*\d+) +(\w+) +(.+)$/
```

Comment:
also please add in comment above what is expected format you want to match as regexp are always a bit tricky and if output changed, then it is easy to see that it no longer matches expectation.
Also it is not clear for me why you do not use `\s` as it is not allowed in output and if systemctl start using tabs, then it is extremely hard to debug it as it is not easy to see.